staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -604,9 +615,22 @@ func (r *Request) WatchWithSpecificDecoders(wrapperDecoderFn func(io.ReadCloser)
 		}
 		return nil, fmt.Errorf("for request %s, got status: %v", url, resp.StatusCode)
 	}
-	wrapperDecoder := wrapperDecoderFn(resp.Body)
+
+	contentType := resp.Header.Get("Content-Type")
```

Comment:
This content type will have `;stream=watch` for proto watch streams, which we'll parse to params, and ignore in the StreamDecoder impl. Just pointing that out.